src/WebJobs.Script/Utility.cs

```diff
@@ -34,6 +35,13 @@ public static class Utility
 
         private static readonly string UTF8ByteOrderMark = Encoding.UTF8.GetString(Encoding.UTF8.GetPreamble());
         private static readonly FilteredExpandoObjectConverter _filteredExpandoObjectConverter = new FilteredExpandoObjectConverter();
+        private static readonly string[] _allowedFunctionNameKeys = new[]
+        {
+            "functionName",
+            LogConstants.NameKey,
+            ScopeKeys.FunctionName
+        };
```

Comment:
If the category of the logger is not already indicating that it is a function, we'll look through the log state, then the log scope, and if we find one of these values, we'll use it as the `FunctionName`. First one we find, wins.